### PR TITLE
Archive periods

### PR DESCRIPTION
**Pull request by Dantemss**
Depends on https://github.com/openstax/tutor-server/pull/1107

- Restore period API
- Specs for deleting/restoring
- Made students in archived periods not count as active
- Added archived periods to the course roster

### CODE REVIEW

File: app/routines/collect_course_info.rb [hunk 1]
```diff
@@ -24,8 +24,6 @@ def collect_courses(courses: nil, user: nil)
   end
 
   def collect_course_info(courses, user, with)
-    with = enforce_order(of: :roles, before: :students, within: with)
```
Comment (jpslav) [inline app/routines/collect_course_info.rb]: Why did this go away?

Comment (jpslav) [inline app/routines/collect_course_info.rb]: Maybe because getting course periods now takes the roles...

**Comment by jpslav** [comment]
Looks good. Does this need to simul-merge with a tutor-js PR?

File: app/routines/collect_course_info.rb (hunk 1 above)
Comment (Dantemss) [inline app/routines/collect_course_info.rb]: I changed the methods so enforcing the order of the method calls was not needed.

File: app/routines/collect_course_info.rb [hunk 2]
```diff
@@ -79,47 +73,34 @@ def set_teacher_names(info, entity_course)
...
-    else
-      student_roles = run(:get_course_roles, course: entity_course, user: user, types: :student)
-                        .outputs.roles
-      student_period_ids = student_roles.map{ |role| role.student.period.id }.uniq
-      info.periods = periods.select{ |period| student_period_ids.include?(period.id) }
-    end
+    @roles ||= run(:get_course_roles, course: entity_course, user: user).outputs.roles
+
+    info.periods = run(:get_course_periods, course: entity_course, roles: @roles,
+                                            include_archived: true).outputs.periods
```
Comment (nathanstitt) [inline app/routines/collect_course_info.rb]: @Dantemss This is including archived periods in the teacher's course JSON.  Is that intended?  FE can filter them out easily enough, but I thought you'd mentioned that archived periods should not sent.

Comment (Dantemss) [inline app/routines/collect_course_info.rb]: I think you are right. I'll remove.

Comment (Dantemss) [inline app/routines/collect_course_info.rb]: Actually I think I might not be able to. See slack.

**Pull request merged**